### ISSUE TITLE
subscribe won't work when first connect fail

### ISSUE THREAD

**Issue by tinpont** [laravel/echo#83]
```
            window.Echo.channel('chat-room.1')
                .listen('ChatMessageWasReceived', (e) => {
                    console.log(e.user, e.chatMessage);
                });
```

These code will not work when connect fail, maybe we should resend `subscribe` to all events when reconnect success. Or offer `connect`, `reconnect`, `error` events to let people handle themselves.

**Comment by shigg1120** [comment]
Yes. It would be necessary to listen for like "connection not established" event or something.

**Comment by driesvints** [comment]
Closing this issue because it's already solved, old or not relevant anymore. Feel free to reply if you're still experiencing this issue.
